File: 1524-Number of Sub-arrays With Odd Sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        running = 0
        odd, even = 0, 1
        kMod = 10**9+7
        res = 0
        
        for num in arr:
            running += num
            if running%2 == 0:
                res = (res + odd)%kMod
            else:
                res = (res + even)%kMod
            
            if running%2 == 0:
                even += 1
            else:
                odd += 1
        
        return res

File: C++/1524-Number of Sub-arrays With Odd Sum.cpp
class Solution {
public:
    int numOfSubarrays(vector<int>& arr) {
        int even = 1, odd = 0, curr = 0, res = 0, kMod = 1e9+7;
        for (int i=0; i<arr.size(); i++) {
            curr += arr[i];
            if (curr%2) {
                res += even;
                odd++;
            } else {
                res += odd;
                even++;
            }
            res%=kMod;
        }

        return res;
    }
};